### PR TITLE
validate premium to set block_list_emails=True

### PR DESCRIPTION
**Pull request by groovecoder**
- [x] I've added a unit test to test for potential regressions of this bug (or this is a front-end change, where we don't yet have unit test infrastructure).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: emails/tests/models_tests.py [hunk 1]
```diff
@@ -237,6 +238,23 @@ def test_valid_address_dupe_of_deleted_invalid(self):
...
         )
 
+    def test_free_user_cant_set_block_list_emails(self):
+        relay_address = RelayAddress.objects.create(user=self.user)
+        try:
+            relay_address.block_list_emails = True
+            relay_address.save()
+        except ValidationError:
+            relay_address.refresh_from_db()
+            assert relay_address.block_list_emails == False
```
Comment (jwhitlock) [inline emails/tests/models_tests.py:248]: This test will pass if the exception is not raised. Please use `assertRaises` to ensure `ValiationError` is raised:

```suggestion
        relay_address.block_list_emails = True
        with self.assertRaises(ValidationError):
            relay_address.save()
        relay_address.refresh_from_db()
        assert relay_address.block_list_emails == False
```

**Review by jwhitlock — CHANGES_REQUESTED** [review]
Looks good, thanks @groovecoder! One change needed to an existing test, and the suggested change to the new test.

I'm seeing this test fail locally and in CI:

```
emails/tests/views_tests.py::SNSNotificationTest::test_block_list_email_sns_notification
```

`self.ra` is associated with a free user, some more work is needed to upgrade the user or associate it with ``self.premium_user ``:

https://github.com/mozilla/fx-private-relay/blob/c6fbdd123683c9fe614e581bfa94c80a43fee47f/emails/tests/views_tests.py#L99-L103

There is an existing API validator ``PremiumValidatorsMixin``, so this won't have the same problem as issue #1558 when using the API.

**Comment by groovecoder** [comment]
Updated to use `SuspiciousOperation` to cause a 4xx error instead of 5xx, and fixed the test.

**Comment by jwhitlock** [comment]
I'm not a fan of using [SuspicousOperation](https://docs.djangoproject.com/en/4.0/ref/exceptions/#suspiciousoperation). It does return a 400, but it also logs at Error level, because it usually means someone is doing something that means they are trying to get around security. `PermissionDenied` or `ValidationError` seems right, but some additional work would be needed to return the 400 instead of a 50x.

Do we have time to move the check higher in the application logic, and make this more like a "assert this never happens" check? Or does this need to go for the sprint, and we can refactor later?

**Comment by groovecoder** [comment]
I don't think this change is critical to this release - the only code-path to change this field goes thru the API which already has a validator to make sure the user is premium.

Sorry I didn’t include it in code comments or even [the other PR](https://github.com/mozilla/fx-private-relay/pull/1429) where I introduced `SuspiciousOperation` for 4xx status code exceptions. But I looked up some different ways to change exceptions to return 4xx status codes. It started to get messy and hack-y to the point that I thought it could cause more problems than it would solve for us. And then I saw that when we update to Django 3.2+ we’ll get to change these to [`BadRequest` exceptions](https://docs.djangoproject.com/en/4.0/ref/exceptions/#badrequest), which gives us a better-matching 4xx exception.

So basically:

1. I don't think this needs to land for the release - it can follow the initial release
2. We can do additional work to return 4xx status codes some other way, or we can defer that to our Django 3.2+ work

**Review by jwhitlock — APPROVED** [review]
The distinction I'd like to make is "did the user do something wrong, or do we have a bug?" If the user did something wrong, a Bad Request response with a 400 status code is the right thing. If we have a bug, I could go with a 40x or 50x, but I'd want to see it in Sentry.

This PR meets my criteria - if the code gets all the way to the save method with this invalid combination, then ``SuspiciousOperation`` will log at Error level, which currently shows up in Sentry. The user gets a 400, which is also consistent with the same error from the API.

We may adjust logging error responses with issue #1556. Other stuff like invalid host headers show up as ``SuspiciousOperation``, and appear as events in Sentry that we can't do anything about. And, we can adjust with Django 3.2+. So, there are opportunities to readjust our strategy.

I'll leave it up to you if you want to get this in for the release. It seems safer to have it, to preserve data consistency, then to leave it out.

**Comment by groovecoder** [comment]
I'll merge it so it can ride the QA fix trains and go thru QA cycle while we're at it.
